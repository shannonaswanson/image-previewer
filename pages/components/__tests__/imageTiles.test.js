import { screen, render, cleanup } from "@testing-library/react";
import renderer from 'react-test-renderer';
import ImageTiles from '../imageTiles';

describe('ImageTiles', () => {
    const images = ['hello', 'world'];
    const sourceDirectory = '/source/directory';

    afterEach(() => cleanup());

    test('should render without crashing', () => {
        render(<ImageTiles />);
    });

    test('should match snap shot', () => {
        const jsonTree = renderer.create(<ImageTiles images={images} sourceDirectory={sourceDirectory} />).toJSON();
        expect(jsonTree).toMatchSnapshot();
    });

    test('should render 2 images', () => {
        render(<ImageTiles images={images} sourceDirectory={sourceDirectory} />);
        const imageContainerElement1 = screen.getByTestId('image-container-0');
        expect(imageContainerElement1).toBeInTheDocument();

        const imageContainerElement2 = screen.getByTestId('image-container-1');
        expect(imageContainerElement2).toBeInTheDocument();
    })
});