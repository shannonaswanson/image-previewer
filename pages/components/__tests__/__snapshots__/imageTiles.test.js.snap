// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageTiles should match snap shot 1`] = `
Array [
  <div
    data-testid="image-container-0"
  >
    <div
      data-testid="image-tile"
    >
      <h3
        className="h3__ImageTitle-sc-2ay7y7-0 hwAtaM"
        data-testid="image-title"
      >
        hello
      </h3>
      <img
        className="imageContainer__ImageContainer-sc-139ry6z-0 cvbhVk"
        data-testid="image-container"
        src="/source/directory/hello"
      />
    </div>
  </div>,
  <div
    data-testid="image-container-1"
  >
    <div
      data-testid="image-tile"
    >
      <h3
        className="h3__ImageTitle-sc-2ay7y7-0 hwAtaM"
        data-testid="image-title"
      >
        world
      </h3>
      <img
        className="imageContainer__ImageContainer-sc-139ry6z-0 cvbhVk"
        data-testid="image-container"
        src="/source/directory/world"
      />
    </div>
  </div>,
]
`;
